src/plot.cpp

```diff
@@ -1582,6 +1532,273 @@ void ProjectionPlot::set_field_of_view(pugi::xml_node node)
   }
 }
 
+PhongPlot::PhongPlot(pugi::xml_node node) : RayTracePlot(node)
+{
+  set_opaque_ids(node);
+  set_diffuse_fraction(node);
+  set_light_position(node);
+}
+
+void PhongPlot::print_info() const
+{
+  fmt::print("Plot Type: Phong\n");
+  RayTracePlot::print_info();
+}
+
+void PhongPlot::create_output() const
+{
+  size_t width = pixels_[0];
+  size_t height = pixels_[1];
+  ImageData data({width, height}, not_found_);
+
+#pragma omp parallel
+  {
+
+#ifdef _OPENMP
+    const int n_threads = omp_get_max_threads();
+    const int tid = omp_get_thread_num();
+#else
+    int n_threads = 1;
+    int tid = 0;
+#endif
+
+    int vert = tid;
+    for (int iter = 0; iter <= pixels_[1] / n_threads; iter++) {
+      if (vert < pixels_[1]) {
+        for (int horiz = 0; horiz < pixels_[0]; ++horiz) {
+
+          // RayTracePlot implements camera ray generation
+          std::pair<Position, Direction> ru = get_pixel_ray(horiz, vert);
+          PhongRay ray(ru.first, ru.second, *this);
```

Comment:
Funky.

Reply:
In what direction would you prefer to change this? The camera-like class here will have to provide pairs of starting positions and directions for rays to follow, so this seemed pretty natural to me.

Are you thinking that `get_pixel_ray` should return an observing pointer to the `Ray` class, or something like that?

Reply:
Oh, do you mean passing `*this`? I think it makes sense. The ray needs to observe some options such as coloring data from the plot class that built it, so it gets an observing reference from `*this`. Otherwise you'd have to copy a bunch of the options like material colors into the ray, which doesn't make a lot of sense.

Is there another way to do it?